File: JavaScript/containsNearbyDuplicate.js


let nums = [1,2,3,1] 
let k = 3


var containsNearbyDuplicate = function(nums, k) {
    
    let seen = {}

    for (x = 0; x < nums.length; x++) {

        if (seen.hasOwnProperty(nums[x]) && Math.abs(x - seen[nums[x]]) <= k) return true

        seen[nums[x]] = x
    
    }
    return false




};


console.log(containsNearbyDuplicate(nums, k))

File: Python/containsNearbyDuplicate.py

nums = [1,1,3,1,2,3]
k = 2

def containsNearbyDuplicate(nums, k):
   
    seen = {}

     for y in range(0, len(nums)):
        
        if (nums[y] in seen and abs(y - seen[nums[y]]) <= k):
            return True
        seen[nums[y]] = y 
        print(seen)
    return False
   
print(containsNearbyDuplicate(nums, k))